### ISSUE TITLE
Translation of stream language

### ISSUE THREAD

**Issue by baka0815** [jellyfin/jellyfin#12821]
The language of streams is - as far as I can tell - not properly translated.

I have subtitle streams which are displayed as "Englisch" (correct translation of en/eng/English) and some which are displayed as "Ger" which should be "Deutsch" (German).

Trying to find the source I found this line:
https://github.com/jellyfin/jellyfin/blob/0bc142de9acd01b8b9191b8e99da41fdebabe0a7/MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs#L716

Is the translation missing or am I looking at the wrong place?

**Comment by baka0815** [comment]
I'm willing to look into this myself, however I would need some direction.

**Comment by IceStormNG** [comment]
I found the actual problem. The issue here:

https://github.com/jellyfin/jellyfin/blob/0bc142de9acd01b8b9191b8e99da41fdebabe0a7/MediaBrowser.Model/Entities/MediaStream.cs#L374-L379

The official three letter code for German is "DEU" but jellyfin uses "Ger" sometimes (or receives that from ffmpeg), therefore, the CultureInfo of .Net will not translate it.

There might be a lookup table needed for the few codes that do not match properly.

If add something like this for debug purpose there:

```csharp
if (Language.Equals("ger", StringComparison.OrdinalIgnoreCase))
{
    fullLanguage = "German";
}
````

The subtitle (in this case) shows up properly as "German" in Jellyfin Web.

Though, I wouldn't hardcode the values here, but create some LUT to translate such values. There might be other country codes that are affected by this. But I also noticed the issue with German Subtitles, especially when external MKS files.

**Comment by baka0815** [comment]
This seems to be the problem described in the comment in your code snippet:
``// Get full language string i.e. eng -> English. Will not work for some languages which use ISO 639-2/B instead of /T codes.``

According to Wikipedia (https://de.wikipedia.org/wiki/Liste_der_ISO-639-2-Codes) ``ger`` is from the /B list.

So we might create a dictionary containing the ``/B from /T``-mapping?

**Comment by IceStormNG** [comment]
Yes. And then is the question, do we translate it here for the display name, or do we translate the values already in the line which you noted, where the language is read from the files.

**Comment by nielsvanvelzen** [comment]
If you just need to build the server you can use the dotnet CLI, just run `dotnet build` / `dotnet publish` to build the code. An IDE like Rider/Visual Studio (code) also supports building the server.

**Comment by baka0815** [comment]
I think removing the "stale" label is enough to stop the bot from closing the issue within 21 days, however I think posting a comment that work on this is still ongoing (see #13068) can't hurt.
